### PR TITLE
refactor: add lightweight styling resources for ToggleSwitch control

### PR DESCRIPTION
**Pull request by Arieldelossantos**
﻿GitHub Issue: [Lightweight styling - ToggleSwitch](https://github.com/unoplatform/uno.csharpmarkup/issues/243)

## PR Type

What kind of change does this PR introduce?

- Refactoring (no functional changes, no api changes)

## Description

add lightweight styling resources for ToggleSwitch control

## PR Checklist 
Please check if your PR fulfills the following requirements:

- Commits must be following the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [ ] Tested UWP
- [ ] Tested iOS
- [ ] Tested Android
- [ ] Tested WASM
- [ ] Tested MacOS
- [x] Contains **No** breaking changes
  > If the pull request contains breaking changes, commit message must contain a detailed description of the action to take for the consumer of this library. As explained by the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: src/library/Uno.Material/Styles/Controls/v2/ToggleSwitch.xaml [hunk 1]
```diff
@@ -36,31 +13,142 @@
...
+			<Thickness x:Key="SwitchKnobOffShadowMargin">0,0,20,0</Thickness>
+		</ResourceDictionary>
+		<ResourceDictionary x:Key="Light">
+			<!-- MaterialToggleSwitchStyle -->
+			<StaticResource x:Key="ToggleSwitchOffOuterBorderFill" ResourceKey="SurfaceVariantBrush" />
+			<StaticResource x:Key="ToggleSwitchOffOuterBorderStroke" ResourceKey="OutlineBrush" />
+			<StaticResource x:Key="ToggleSwitchOuterBorderStrokeNormal" ResourceKey="OutlineBrush" />
+			<StaticResource x:Key="ToggleSwitchOuterBorderFillNormal" ResourceKey="SurfaceVariantBrush" />
+			<StaticResource x:Key="ToggleSwitchKnobOffFill" ResourceKey="OnSurfaceVariantBrush" />
+			<StaticResource x:Key="ToggleSwitchKnobOffFillNormal" ResourceKey="OnSurfaceVariantBrush" />
```
Comment (kazo0) [inline src/library/Uno.Material/Styles/Controls/v2/ToggleSwitch.xaml:89]: The "Normal" VisualState is a special case where we don't need a separate resource with "Normal" attached at at the. In the Normal VisualState we should just be using "ToggleSwitchKnobOffFill" so there's no need to have two resources here with the same value, they can be merged into one resource without the "Normal" at the end end of the name

**Review by Xiaoy312 — APPROVED** [review]

**Review by agneszitte — APPROVED** [review]

**Review by agneszitte — APPROVED** [review]

**Review by kazo0 — COMMENTED** [review]
@Arieldelossantos the UWP sample app seems to be crashing on startup with these changes

File: src/library/Uno.Material/Styles/Controls/v2/ToggleSwitch.xaml [hunk 2]
```diff
@@ -3,30 +3,122 @@
 					xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
```
Comment (kazo0) [inline src/library/Uno.Material/Styles/Controls/v2/ToggleSwitch.xaml:3]: @Arieldelossantos the UWP sample app seems to be crashing on startup with these changes

Comment (Arieldelossantos) [reply src/library/Uno.Material/Styles/Controls/v2/ToggleSwitch.xaml:4]: problem solved, the issue was caused for using ThemeResources on a converter... for some reason it crash... 
for now what i did was extracted the 2 resources the converter needs and kept using staticResource for only those 2

**Review by kazo0 — APPROVED** [review]

**Comment by kazo0** [comment]
https://github.com/Mergifyio backport legacy/2x

**Pull request merged**